[BUG] @playwright/test type errors using latest TypeScript nightly

**Context:**
- Playwright Version: `@playwright/test` `1.20.1`
- Operating System: MacOS
- Node.js version: 16
- Browser: NA
- Extra: NA

 
<!-- CLI to auto-capture this info -->
<!-- npx envinfo --preset playwright --markdown -->

**Repo**
1. Install latest TS nightly build (tested with `4.7.0-dev.20220329`)
2. Install `@playwright/test` 1.20.1
3. In a TS file with strict null checks enabled, write:

```ts
import * as playwright from '@playwright/test';
```

4. Try compiling 

**Describe the bug**

This produces the following compile errors:

```
matb@mjbmbp16 san % npx tsc -p tsconfig.json
node_modules/@playwright/test/types/test.d.ts:2682:57 - error TS2344: Type 'T' does not satisfy the constraint 'KeyValue'.

2682   extend<T, W extends KeyValue = {}>(fixtures: Fixtures<T, W, TestArgs, WorkerArgs>): TestType<TestArgs & T, WorkerArgs & W>;
                                                             ~

  node_modules/@playwright/test/types/test.d.ts:2682:10
    2682   extend<T, W extends KeyValue = {}>(fixtures: Fixtures<T, W, TestArgs, WorkerArgs>): TestType<TestArgs & T, WorkerArgs & W>;
                  ~
    This type parameter probably needs an `extends object` constraint.

node_modules/@playwright/test/types/test.d.ts:2693:72 - error TS2344: Type 'Args' does not satisfy the constraint 'KeyValue'.

2693 type TestFixtureValue<R, Args> = Exclude<R, Function> | TestFixture<R, Args>;
                                                                            ~~~~

  node_modules/@playwright/test/types/test.d.ts:2693:26
    2693 type TestFixtureValue<R, Args> = Exclude<R, Function> | TestFixture<R, Args>;
                                  ~~~~
    This type parameter probably needs an `extends object` constraint.

node_modules/@playwright/test/types/test.d.ts:2694:76 - error TS2344: Type 'Args' does not satisfy the constraint 'KeyValue'.

2694 type WorkerFixtureValue<R, Args> = Exclude<R, Function> | WorkerFixture<R, Args>;
                                                                                ~~~~

  node_modules/@playwright/test/types/test.d.ts:2694:28
    2694 type WorkerFixtureValue<R, Args> = Exclude<R, Function> | WorkerFixture<R, Args>;
                                    ~~~~
    This type parameter probably needs an `extends object` constraint.


Found 3 errors in the same file, starting at: node_modules/@playwright/test/types/test.d.ts:2682
```

This may be another case of https://github.com/microsoft/TypeScript/issues/48468

--- a/packages/playwright-test/types/test.d.ts
+++ b/packages/playwright-test/types/test.d.ts
@@ -2819,7 +2819,7 @@ export interface TestType<TestArgs extends KeyValue, WorkerArgs extends KeyValue
    * Learn more about [fixtures](https://playwright.dev/docs/test-fixtures) and [parametrizing tests](https://playwright.dev/docs/test-parameterize).
    * @param fixtures An object containing fixtures and/or options. Learn more about [fixtures format](https://playwright.dev/docs/test-fixtures).
    */
-  extend<T, W extends KeyValue = {}>(fixtures: Fixtures<T, W, TestArgs, WorkerArgs>): TestType<TestArgs & T, WorkerArgs & W>;
+  extend<T extends KeyValue, W extends KeyValue = {}>(fixtures: Fixtures<T, W, TestArgs, WorkerArgs>): TestType<TestArgs & T, WorkerArgs & W>;
   /**
    * Returns information about the currently running test. This method can only be called during the test execution,
    * otherwise it throws.
@@ -2830,8 +2830,8 @@ export interface TestType<TestArgs extends KeyValue, WorkerArgs extends KeyValue
 type KeyValue = { [key: string]: any };
 export type TestFixture<R, Args extends KeyValue> = (args: Args, use: (r: R) => Promise<void>, testInfo: TestInfo) => any;
 export type WorkerFixture<R, Args extends KeyValue> = (args: Args, use: (r: R) => Promise<void>, workerInfo: WorkerInfo) => any;
-type TestFixtureValue<R, Args> = Exclude<R, Function> | TestFixture<R, Args>;
-type WorkerFixtureValue<R, Args> = Exclude<R, Function> | WorkerFixture<R, Args>;
+type TestFixtureValue<R, Args extends KeyValue> = Exclude<R, Function> | TestFixture<R, Args>;
+type WorkerFixtureValue<R, Args extends KeyValue> = Exclude<R, Function> | WorkerFixture<R, Args>;
 export type Fixtures<T extends KeyValue = {}, W extends KeyValue = {}, PT extends KeyValue = {}, PW extends KeyValue = {}> = {
   [K in keyof PW]?: WorkerFixtureValue<PW[K], W & PW> | [WorkerFixtureValue<PW[K], W & PW>, { scope: 'worker', timeout?: number | undefined }];
 } & {

--- a/utils/generate_types/overrides-test.d.ts
+++ b/utils/generate_types/overrides-test.d.ts
@@ -339,15 +339,15 @@ export interface TestType<TestArgs extends KeyValue, WorkerArgs extends KeyValue
   use(fixtures: Fixtures<{}, {}, TestArgs, WorkerArgs>): void;
   step(title: string, body: () => Promise<any>): Promise<any>;
   expect: Expect;
-  extend<T, W extends KeyValue = {}>(fixtures: Fixtures<T, W, TestArgs, WorkerArgs>): TestType<TestArgs & T, WorkerArgs & W>;
+  extend<T extends KeyValue, W extends KeyValue = {}>(fixtures: Fixtures<T, W, TestArgs, WorkerArgs>): TestType<TestArgs & T, WorkerArgs & W>;
   info(): TestInfo;
 }
 
 type KeyValue = { [key: string]: any };
 export type TestFixture<R, Args extends KeyValue> = (args: Args, use: (r: R) => Promise<void>, testInfo: TestInfo) => any;
 export type WorkerFixture<R, Args extends KeyValue> = (args: Args, use: (r: R) => Promise<void>, workerInfo: WorkerInfo) => any;
-type TestFixtureValue<R, Args> = Exclude<R, Function> | TestFixture<R, Args>;
-type WorkerFixtureValue<R, Args> = Exclude<R, Function> | WorkerFixture<R, Args>;
+type TestFixtureValue<R, Args extends KeyValue> = Exclude<R, Function> | TestFixture<R, Args>;
+type WorkerFixtureValue<R, Args extends KeyValue> = Exclude<R, Function> | WorkerFixture<R, Args>;
 export type Fixtures<T extends KeyValue = {}, W extends KeyValue = {}, PT extends KeyValue = {}, PW extends KeyValue = {}> = {
   [K in keyof PW]?: WorkerFixtureValue<PW[K], W & PW> | [WorkerFixtureValue<PW[K], W & PW>, { scope: 'worker', timeout?: number | undefined }];
 } & {

